### ISSUE TITLE
Filtered attributes on static model methods

### ISSUE THREAD

**Issue by cookrn** [RobinBuschmann/sequelize-typescript#553]
When attempting to use static methods to create (or `update`), it does not seem that the `FilteredModelAttributes<Account>` type for the argument is executing as expected. For example:

```typescript
import {
  AllowNull,
  Column,
  DataType,
  Default,
  IsEmail,
  IsUUID,
  Model,
  PrimaryKey,
  Table,
} from "sequelize-typescript";

@Table
export class Account extends Model<Account> {
  @IsUUID(4)
  @PrimaryKey
  @Default(DataType.UUIDV4)
  @Column
  public id!: string;

  @AllowNull(false)
  @Column
  public firstName!: string;

  @AllowNull(false)
  @Column
  public lastName!: string;

  @AllowNull(false)
  @IsEmail
  @Column
  public email!: string;

}

const input = {
  email: "email@example.com",
  firstName: "Ryan",
  lastName: "Cook",
  zoology: "wut"
};

const account = await Account.create(input);  // <-- this line doesn't complain about the zoology key
```

Do I need to explicitly specify the type parameters to those static methods or are they implied? For example, if I copy in the relevant types from `Model.d.ts` and utils and write the following:

```typescript
const account = await Account.create<Account, FilteredModelAttributes<Account>>(input);
```

That still doesn't provide an error for the `zoology` key, but the following does error:

```typescript
const input: FilteredModelAttributes<Account> = {
  email: "email@example.com",
  firstName: "Ryan",
  lastName: "Cook",
  zoology: "wut" // <-- this line complains now that zoology isn't valid on Account
};
```

I appreciate any input into what I may be misunderstanding!

**Comment by cookrn** [comment]
This is possibly related to #536

**Comment by RobinBuschmann** [comment]
Hey @cookrn, thanks for reporting. This is because of the second line of the `create` typings, which allows any object to get passed. See:
```ts
 static create<T extends Model<T>>(this: (new () => T), values?: FilteredModelAttributes<T>, options?: ICreateOptions): Promise<T>;
  static create<T extends Model<T>, A>(this: (new () => T), values?: A, options?: ICreateOptions): Promise<T>; // <--
```
We could remove this in order to be more restrictive here. What do you think?

(Please note, that `sequelize-typescript@1.0.0` will use the typings provided by sequelize itself, where most of the typings aren't as strict as they are now.)

**Comment by cookrn** [comment]
Good question! What is the use-case for allowing any object to be passed versus checking if the values are of the required type?

**Comment by RobinBuschmann** [comment]
I think, I didn't want to break something. The thing is, that the exact type isn't documented. See the [create documentation on sequelize](http://docs.sequelizejs.com/class/lib/model.js~Model.html#static-method-create).

So with the current approach we are getting auto completion, but doesn't really restrict what gets passed.

**Comment by cookrn** [comment]
Understood. Since we have the ability to create our own internal interfaces in our application which can be more strictly typed, I think your decision makes sense. (The decision being to support autocompletion but not strict argument checking.) Unless you'd like to move forward with changing something, I think this issue can be closed. Thank you so much for discussing!

**Comment by RobinBuschmann** [comment]
You‘re welcome
